src/ball_tree.rs

```diff
@@ -189,8 +177,7 @@ where
             let (min_i, min_dist) = self.idx[root_node.range.clone()].iter().fold(
                 (0, A::infinity()),
                 |(min_i, min_dist), &i| {
-                    let distance = self.distance;
-                    let dist = distance(point, &self.points.row(i));
+                    let dist = self.distance.get(point, &self.points.row(i));
```

Comment:
Since `distance` is no longer directly callable, how about naming it `metric` so that we can call `metric.distance()`, rather than `distance.get()`? I think `get` is too general.